### ISSUE TITLE
Lack of description of logging via connect

### ISSUE THREAD

**Issue by Make42** [allegroai/clearml-docs#473]
It took me a whole day to figure out what was wrong with my logging. Finally, I was able to resolve it thanks to someone in the forum pointing to https://github.com/allegroai/clearml/blob/master/examples/reporting/hyper_parameters.py and a lot more testing until I understood my mistake.

Here a minimal working example:

I had the code:
```
from clearml import Task
parameters = {'var': 1.0}
task = Task.init(project_name='MyProject', task_name='my_task')
task.connect(parameters)
parameters ['var'] = 2.0
```
but correct is
```
from clearml import Task
parameters = {'var': 1.0}
task = Task.init(project_name='MyProject', task_name='my_task')
parameters = task.connect(parameters)
parameters ['var'] = 2.0
```

Difficult to spot and also unexpected because "connect" sounds like a method, which side-effect does the work. Because I expected [command-query separation](https://en.wikipedia.org/wiki/Command%E2%80%93query_separation) I would think that the return value is not important. Finally, the documentation https://clear.ml/docs/latest/docs/references/sdk/task#connect says about the return value with

> Return type: Any
> Returns: The result returned when connecting the object, if supported.

basically nothing, which further supported my way of thinking that the return value is not important.

Finally, the docstring is also confusing regarding what objects are allowed. First it lists a whole bunch, but then says, only dict and `TaskParameters` is supported. Which is it? Or is the long list the future plan and only dict and `TaskParameters` are implemented for now? Please make this explicit as well.

Long story short: Please make the web documentation more explicit, as well as the docstrings, give examples in both cases (directly at the respective places).

**Comment by ainoam** [comment]
Appreciate the feedback @Make42 - We'll make sure to fix the docstrings and clarify the docs.

**Comment by pollfly** [comment]
Hey @Make42! The docstrings for [`Task.connect`](https://clear.ml/docs/latest/docs/references/sdk/task#connect) have been fixed.

**Comment by jkhenning** [comment]
@Make42 closing this, please reopen if required.

**Comment by Make42** [comment]
@jkhenning @pollfly : I just read [the docstring](https://clear.ml/docs/latest/docs/references/sdk/task/#connect). I would not be able to understand what I need to do from reading it.
